files/default/hopsworks_templates/jupyter_notebook_config_template.py

```diff
@@ -70,3 +64,29 @@
 
 os.environ['HADOOP_CLASSPATH_GLOB'] = "${conf.hadoopClasspathGlob}"
 os.environ['HADOOP_LOG_DIR'] = "/tmp"
+
+# hopsfs-mount does not support xattr. If we save the file then we loose the xattr information of the notebook
+# We need to put back the xattr after saving the notebook.
+import concurrent.futures
+notebook_save_thread_pool = concurrent.futures.ThreadPoolExecutor(max_workers=7)
+
+def notebook_post_save_hook(model, os_path, contents_manager, **kwargs):
+    # only run on notebooks
+    if model['type'] != 'notebook':
+        return
+    if os.environ['JUPYTER_HOPSFS_MOUNT']:
+        try:
+            import sys
+            import threading
+            from hops import util
+        except Exception as err:
+            return
+        if 'hops.util' in sys.modules:
```

Comment:
I would add everything inside the `try...except` If the lines below raise any exception they will be propagated to Jupyter server